build-system/runner/src/org/ampproject/AmpCommandLineRunner.java

```diff
@@ -87,6 +87,9 @@ protected AmpCommandLineRunner(String[] args) {
     options.setRemoveUnusedPrototypeProperties(false);
     options.setInlineProperties(false);
     options.setComputeFunctionSideEffects(false);
+    // Since we are not computing function side effects, at least let the
```

Comment:
Does anyone remember why the above is false?

Reply:
Flip it tooo!

Reply:
Hmm, I suppose if the base JS has a class with a method that has no side effects (not annotated as such), but then a subclass in a different compilation unit overrides it and has side effects, then if you have a reference to the parent type and call the method, then you cannot safely dead-code eliminate it because the function you are actually calling may have sideffects.

I think `@nosideeffects` should be safe since it can only appear in externs and developers can't add it themselves.

Reply:
Yes, that might have been the reason I turned it off :) And I agree `@nosideeffects` is safe.